src/containers/ListContainer.js

```diff
@@ -10,141 +10,215 @@ import { Pressable } from 'react-native';
 import MagnifyingIcon from '../assets/images/icons/magnifying.png';
 import Selection from '../assets/images/icons/selection.png';
 import Sort from '../assets/images/icons/sort.png';
+import XIcon from '../assets/images/icons/x-icon.png';
+import SearchBar from '../components/SearchBar.js';
 
 class ListContainer extends Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            isToggled: false
+            isToggled: false,
+            className: '',
+            classEntered: false,
+            nameEntered: false,
+            showLabel: false,
+            isReadOnly: false,
+            selectedButton: 0,
+            participantName: ''
         };
 
-        this.onToggle = this.onToggle.bind(this)
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.filterBarVisibility = this.filterBarVisibility.bind(this);
+        this.deleteLabel = this.deleteLabel.bind(this);
+        this.searchBarVisibility = this.searchBarVisibility.bind(this);
     }
 
-    onToggle() {
-        this.setState({isToggled: !this.state.isToggled});
-    }
+    deleteLabel = () => {
+        this.setState({ showLabel: false });
+        console.log("Deleted");
+      }
+    
+      handleSubmit = (event) => {
+        this.setState({ showLabel: true });
+        this.setState({ className: event.nativeEvent.text })
+        this.className.clear();
+      }
+
+      handleSearchSubmit = () => {
+        // TODO
+        this.participantName.clear();
+
+      }
+      filterBarVisibility = () => {
+        this.setState({ selectedButton: this.state.selectedButton = 1})
```

Comment:
what does this do? `this.setState({ selectedButton: this.state.selectedButton = 1})`
How is it different from `this.setState({ selectedButton: 1})`

Reply:
changed it to boolean